x-pack/plugins/fleet/server/services/epm/packages/get.ts

```diff
@@ -346,36 +354,65 @@ export async function getInstalledPackageManifests(
   savedObjectsClient: SavedObjectsClientContract,
   installedPackages: InstalledPackage[]
 ) {
-  const pathFilters = installedPackages.map((installedPackage) => {
+  // Integration manifests
+  const topLevelManifestPathFilters = installedPackages.map((installedPackage) => {
     const { name, version } = installedPackage;
     return nodeBuilder.is(
       `${ASSETS_SAVED_OBJECT_TYPE}.attributes.asset_path`,
       `${name}-${version}/manifest.yml`
     );
   });
 
-  const result = await savedObjectsClient.find<PackageAsset>({
+  const topLevelIntegrationManifests = await savedObjectsClient.find<PackageAsset>({
     type: ASSETS_SAVED_OBJECT_TYPE,
-    filter: nodeBuilder.or(pathFilters),
+    filter: nodeBuilder.or(topLevelManifestPathFilters),
   });
 
-  const parsedManifests = result.saved_objects.reduce<Map<string, PackageSpecManifest>>(
-    (acc, asset) => {
-      acc.set(asset.attributes.asset_path, yaml.load(asset.attributes.data_utf8));
-      return acc;
-    },
-    new Map()
-  );
+  const parsedTopLevelIntegrationManifests = topLevelIntegrationManifests.saved_objects.reduce<
```

Comment:
**nit**: do you think would make sense to extract this logic into a getManifestsMap that can be used for both top-level integrations and dataStreams later in this function? Just for readability purposes.